### PR TITLE
Allow user-set cannonball count before notifying

### PR DESCRIPTION
**Pull request by LucasChilders**
Open to suggestions on the config names / descriptions. 

Resolves https://github.com/runelite/runelite/issues/10365

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java [hunk 1]
```diff
@@ -25,28 +25,43 @@
...
 public interface CannonConfig extends Config
 {
 	@ConfigItem(
-		keyName = "showEmptyCannonNotification",
-		name = "Empty cannon notification",
-		description = "Configures whether to notify you that the cannon is empty"
+		keyName = "enableCannonNotifications",
+		name = "Enable cannon notifications",
+		description = "Configures whether to notify you when your cannon has X cannonballs remaining.",
+		position = 1
```
Comment (KasparRosin) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]: Why does it need position

Comment (Adam-) [reply runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]: Without a position it sorts of alphabetically. Generally we try to group related options next to each other though.

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java [hunk 2]
```diff
@@ -25,28 +25,43 @@
...
 		return true;
 	}
 
+	@Range(
+		max = 30
+	)
+	@ConfigItem(
+		keyName = "numRemainingBeforeNotification",
+		name = "Number for notification",
+		description = "Configures the of cannonballs remaining before the user should be notified. Regardless of this" +
```
Comment (KasparRosin) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]: Typo. 
I'm guessing it's supposed to be `Configures the number of cannonballs`

Comment (LucasChilders) [reply runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]: thanks - fixed

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java [hunk 3]
```diff
@@ -278,6 +278,12 @@ public void onProjectileMoved(ProjectileMoved event)
 				}
 			}
 		}
+
+		// Don't use this notification if we're at zero, that's handled in the onChatMessage method.
+		if (cballsLeft > 0 && cballsLeft == config.numRemainingBeforeNotification() && config.enableCannonNotifications())
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonPlugin.java]: This needs to be up where `cballsLeft` is decremented.

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java [hunk 4]
```diff
@@ -25,28 +25,43 @@
...
 	{
 		return true;
 	}
 
+	@Range(
+		max = 30
+	)
+	@ConfigItem(
+		keyName = "numRemainingBeforeNotification",
+		name = "Number for notification",
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]: Maybe call this `Low warning threshold`

File: runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java [hunk 5]
```diff
@@ -25,28 +25,43 @@
...
+		description = "Configures the number of cannonballs remaining before the user should be notified. Regardless of this" +
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/cannon/CannonConfig.java]:
```suggestion
		description = "Configures the number of cannonballs remaining before a notification is sent. Regardless of this" +
```

**Comment by Adam-** [comment]
this was combined into #10375

**Pull request closed without merging**